### ISSUE TITLE
string field becomes Object when querying JSON Value

### ISSUE THREAD

**Issue by xiaoas** [SeaQL/sea-orm#1313]
## Description
```rust
JsonValue::find_by_statement(statement).into_json().all(db).await?;
```
This returns a `Vec<serde_json::Value>`, and the value is Value::Object<HashMap<String, Value>>
A string field in the db with json as value becomes an Object in the HashMap.

## Steps to Reproduce
I'll try to see if this issue occurs in sqlite so that I can write a how to reproduce.

### Expected Behavior

The field should be a Value::String

### Actual Behavior

That turns into a Value::Object.
### Reproduces How Often

Yes.
## Versions

```toml
sea-orm = { version = "0.10", features = ["sqlx-mysql", "runtime-tokio-native-tls", "macros", "debug-print"] }
```
I'm using mysql 5.7

**Comment by xiaoas** [comment]
Seems that I cannot reproduce the issue with sqlite🥲

**Comment by xiaoas** [comment]
I tried adding dbg! in src/query/json.rs:30 like this:
```rust
let col_type = column.type_info();
dbg!(column.name());
dbg!(col_type);
macro_rules! match_mysql_type {
```
this mis-interpreted field output is the following:
```rust
[/sea-orm-0.10.2/src/query/json.rs:30] column.name() = "payment_extra"
[sea-orm-0.10.2/src/query/json.rs:31] col_type = MySqlTypeInfo {
    type: Blob,
    flags: BLOB | BINARY,
    char_set: 224,
    max_size: Some(
        262140,
    ),
}
```
The type is blob, I think that's because the field is so long, and that might be the root cause.

**Comment by billy1624** [comment]
Hey @xiaoas, thanks for the investigation and made the filing as discussed on Discord.

https://github.com/SeaQL/sea-orm/blob/f1ae76e130840fc2bc34693e3e3978273d6174c2/src/query/json.rs#L67-L69

The problem is that `into_json` only have the query result, it doesn't have knowledge of the model to determine the exact data type of each column.

I think we can provide a `into_json_with_serializer` method where user can customize the serializer.

```rs
fn into_json_with_serializer<F>(serializer: T) -> Result<serde_json::Value, DbErr>
where
    T: Fn(&QueryResult) -> serde_json::Value
{
    // ...
}
```

I can draft an API this afternoon if I have time :)

**Comment by xiaoas** [comment]
Thanks for the reply😀
> I think we can provide a into_json_with_serializer method where user can customize the serializer.

That won't solve the issue if users does not know about this method, tho. I don't know about other orms but gorm seems to be able to interpret the field as a string, by default.

When the user is asking for a `serde_json::Value` for each record. I think it makes sense that the result will be a flat map: there won't be recursive `serde_json::Value::Object`. `serde_json::Value` as a field value feels unnatural: just like there won't be `chrono::NaiveDateTime` in the result. 

What if we move `try_get_type!(String, col);` upward a little bit? Does that make sense? Or maybe we could throw an error if multiple types passes `try_get_type`?

**Comment by billy1624** [comment]
> That won't solve the issue if users does not know about this method, tho.

Yes, that's another way to let user have full control over the JSON serializer.

> What if we move `try_get_type!(String, col);` upward a little bit? Does that make sense?

Then we will have all nested JSON object to be represented as String.
